pytest_httpserver/httpserver.py

```diff
@@ -675,6 +724,61 @@ def dispatch(self, request: Request) -> Response:
 
         return response
 
+    def _waiting_result_set(self, value: bool) -> None:
+        """Set waiting_result
+
+        Setting is implemented as putting value to queue without waiting. If queue is full we simply ignore the
+        exception, because that means that waiting_result was already set, but not read.
+        """
+        with suppress(queue.Full):
+            self._waiting_result.put_nowait(value)
+
+    def _waiting_result_update(self) -> None:
```

Comment:
This function gets called from two exact places where at each place you can be 100% sure that which list got empty as there's a remove() or the pop() call there (line 710, 704).

I think that the code would be readable if this function would not exist (`_waiting_result_update`) and at those place where it is called there would be a call for `_waiting_result_set(True)`.

Do you think that this `_waiting_result_update` function will be called in the future from more places?

Reply:
Nope. Firstly, after line 704, 710 correspondent list becomes one item smaller, but not definitely empty. Secondly, I want to notify the user when all ordered and oneshot handlers were consumed, not just ordered or oneshot. So, I think everything is fine.

Reply:
Ok, I mis-read the `and` to `or`. Thank for the rename.